### ISSUE TITLE
Azure Provider related WARNING in webserver (version 2.0.1)

### ISSUE THREAD

**Issue by Swalloow** [apache/airflow#14266]
**Apache Airflow version**: 2.0.1

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): 1.18

**Environment**: Official Helm Chart with CeleryExecutor

- **Cloud provider or hardware configuration**: AWS EKS
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

When updated to `apache/airflow:2.0.1-python3.7` image, a WARNING message occurs in the webserver.

```
[2021-02-17 04:06:47,339] {providers_manager.py:299} 
WARNING - Exception when importing 'airflow.providers.microsoft.azure.hooks.wasb.WasbHook' from 
'apache-airflow-providers-microsoft-azure' package: No module named 'azure.storage.blob'
```

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know**:

**Comment by kaxil** [comment]
Try the following command inside the container:

```
pip uninstall  --yes azure-storage && pip install -U azure-storage-blob apache-airflow-providers-microsoft-azure==1.1.0
```

**Comment by JasonPierce** [comment]
Happening in `apache/airflow:2.0.1-python3.8`, as well.

Workaround (thanks @kaxil):

1. Start Airflow: `docker-compose up`
2. Open bash in webserver container: `docker exec -it airflow_airflow-webserver_1 /bin/bash` (use `docker ps` to find your webserver container name)
3. Install azure-storage-blob: `pip uninstall --yes azure-storage && pip install -U azure-storage-blob apache-airflow-providers-microsoft-azure==1.1.0`

Didn't have to restart webserver, it just started working after running steps 1-3. YMMV.

**Comment by don4of4** [comment]
Experiencing this as well -- obviously running commands in the container isn't great.  Is this fixed in a more recent version?

Also seeing: ` Exception when importing 'airflow.providers.google.cloud.hooks.compute_ssh.ComputeEngineSSHHook' from 'apache-airflow-providers-google' package: cannot import name 'ImportSshPublicKeyResponse' from 'google.cloud.oslogin_v1'`

**Comment by potiuk** [comment]
Seems that this issue can be closed. I do not think we can do much more here :(

**Comment by omarish** [comment]
I worked around this by adding another Dockerfile with the pip install:

```dockerfile
FROM apache/airflow:2.0.1
RUN pip uninstall  --yes azure-storage && pip install -U azure-storage-blob apache-airflow-providers-microsoft-azure==1.1.0
```

**Comment by karakanb** [comment]
@potiuk The problem seems to be with the Docker image itself, meaning that there is no dependency installed from the user's perspective. What you are saying would be correct if the user was trying to run them with their local dependencies, but in the case of the provided Docker example there is simply nothing installed from the host device, therefore this needs to be fixed in the image level as far as I can see.

**Comment by potiuk** [comment]
@karakanb -> correct - that was a problem with dependencies in the image. But it will be fixed in the upcoming bugfix releases of the provider - first, and of the airflow version and image (as second step)

Just to clarify the original issue:

The 2.0.1 image was prepared with the provider packages released at that time (`apache-airflow-providers-microsoft-azure` https://pypi.org/project/apache-airflow-providers-microsoft-azure/1.1.0/ - what you can see there is that `azure-storage-blob` was not a dependency of the `microsoft-azure` provider.

This is fixed in the upcoming 1.2.0 provider: https://pypi.org/project/apache-airflow-providers-microsoft-azure/1.2.0rc1/ (released for voting today). You will see there that `azure-storage-blob` is a dependency there.

The way to mitigate the issue:

* install azure-storage-blob manually in your "extended image" -> @omarish's solution works.
* wait until 1.2.0 provider package for `microsoft-azure` gets released (early next week) and install it (similar way as above). You can also install it by specifying the rc version if you would be so kind to test it:

```
FROM apache/airflow:2.0.1
RUN pip install apache-airflow-providers-microsoft-azure==1.2.0rc1
```
*  when we release 2.0.2 version of Airflow (week-two from now I guess), the released packages will be automatically used (including the 1.2.0 of microsoft-azure) so 2.0.2 should not have this problem any more.

Sorry for the confusion. The azure dependencies were quite a bit of a problem (due to old/new libraries used for storage by different providers) - but as of 2.0.2, those problems should be gone forever :crossed_fingers:

**Comment by pl77** [comment]
For those who might be looking for a simple way to implement these suggestions, I tried solutions from both @omarish and @karakanb and they both worked fine.  Easiest way to try them was create a blank text file named, as an example, `Dockerfile-karaknb` and inside place the commands:

```
FROM apache/airflow:2.0.1
RUN pip install apache-airflow-providers-microsoft-azure==1.2.0rc1
```

Then, in the same directory you saved that file, run:

```bash
docker build -f ./Dockerfile-karakanb -t apache/airflow:2.0.1 .
```

If you save with the same tag as the existing file you don't have to change your `docker-compose.yml` files and when 2.0.2 is released it'll supersede this monkey patch when you do your regular `docker-compose pull` when upgrading (of course, that's if your image is specified as `:latest` in your compose file). 

I'm sure everyone using airflow is advanced enough to know this already but figured I'd post it just in case someone was struggling.

And the rc1 candidate seemed to install just fine but I didn't have an azure instance to test it against, unfortunately.
